node/src/database/db_migrations.rs

```diff
@@ -1242,4 +1441,163 @@ mod tests {
+        let start_at = 4;
+        let dir_path = ensure_node_home_directory_exists(
+            "db_migrations",
+            "migration_from_4_to_5_with_pending_transactions",
+        );
+        let db_path = dir_path.join(DATABASE_FILE);
+        let conn = bring_db_0_back_to_life_and_return_connection(&db_path);
+        let wallet_1 = make_wallet("james_bond");
+        let transaction_hash_1 = H256::from_uint(&U256::from(45454545));
+        let wallet_2 = make_wallet("robinson_crusoe");
+        let transaction_hash_2 = H256::from_uint(&U256::from(999888));
+        let subject = DbInitializerReal::default();
+        {
+            let _ = subject
+                .initialize_to_version(
+                    &dir_path,
+                    start_at,
+                    true,
+                    MigratorConfig::create_or_migrate(make_external_migration_parameters()),
+                )
+                .unwrap();
+        }
+        fn prepare_account_with_pending_transaction(
+            conn: &Connection,
+            transaction_hash: H256,
+            wallet: &Wallet,
+            amount: i64,
+            timestamp: SystemTime,
+        ) {
+            let mut stm = conn.prepare("insert into payable (wallet_address, balance, last_paid_timestamp, pending_payment_transaction) values (?,?,?,?)").unwrap();
+            let params: &[&dyn ToSql] = &[
+                &wallet,
+                &amount,
+                &to_time_t(timestamp),
+                &format!("{:x}", transaction_hash),
+            ];
+            let row_count = stm.execute(params).unwrap();
+            assert_eq!(row_count, 1);
+        }
+        prepare_account_with_pending_transaction(
+            &conn,
+            transaction_hash_1,
+            &wallet_1,
+            555555,
+            SystemTime::now(),
+        );
+        prepare_account_with_pending_transaction(
+            &conn,
+            transaction_hash_2,
+            &wallet_2,
+            1111111,
+            from_time_t(200_000_000),
+        );
+        let conn_schema5 = subject
+            .initialize_to_version(
+                &dir_path,
+                start_at + 1,
+                false,
+                MigratorConfig::create_or_migrate(ExternalData::new(
+                    TEST_DEFAULT_CHAIN,
+                    NeighborhoodModeLight::ConsumeOnly,
+                )),
+            )
+            .unwrap();
+
+        assert_on_schema_5_was_adopted(conn_schema5.as_ref());
+        TestLogHandler::new().exists_log_containing("WARN: DbMigrator: Migration from 4 to 5: database belonging to the chain 'eth-ropsten'; \
+         we discovered possibly abandoned transactions that are said yet to be pending, these are: '0000000000000000000000000000000000000000000000000000000002b594d1', '00000000000000000000000000000000000000000000000000000000000f41d0'; continuing");
+    }
+
+    fn assert_on_schema_5_was_adopted(conn_schema5: &dyn ConnectionWrapper) {
```

Comment:
Somewhere--especially if the rename-table-to-old-then-copy-data-to-new-table process is going to stay in place--I'd like to see assertions that old data was actually successfully copied to the new table and still exists.